README.md

```diff
@@ -21,36 +22,45 @@ npm i type-safe-errors
 
 ### Basic example
 
+In this example, we demonstrate how to use the `type-safe-errors` library to handle user authorization with a simple username and password check.
+
 ```ts
 import { Ok, Err } from 'type-safe-errors';
 
 class InvalidCredentialsError extends Error {
   name = 'InvalidCredentialsError' as const;
 }
 
+// Function to authorize a user based on their username and password
 function authorizeUser(username: string, password: string) {
```

Comment:
one of the main point of the library is that types should be always when it possible properly inferred. this can suggest the reader that such return type definition is required to use the library.

Reply:
In Some project part of returning types is mandatory so in my opinion or we should show full example or we can add additional one

Reply:
imo way in which I do not prefer the library be used should be one of first examples in README. we can create some dedicated example or e.g. migrate `nestjs` example to such style in mention and link it somewhere in the main docs.